code/VintageEngineering/Transport/API/BEPipeBase.cs

```diff
@@ -127,12 +129,23 @@ public override void Initialize(ICoreAPI api)
             if (pnm == null) return;
 
             if (numExtractionConnections > 0) 
-            { 
+            {
                 RebuildPushConnections(api.World, pnm.GetNetwork(NetworkID).PipeBlockPositions.ToArray());
                 api.World.BlockAccessor.MarkBlockEntityDirty(Pos);
             }
-            if (api.Side == EnumAppSide.Server) MarkDirty(true);
-        }        
+            if (api.Side == EnumAppSide.Server)
+            {
+                // MagicNum.ServerChunkSize always = 32 some of the game code don't use MagicNum and just use 32 so IDk if it is usefull
+                BlockPos inChunkPos = new (Pos.X%MagicNum.ServerChunkSize, Pos.Y%MagicNum.ServerChunkSize, Pos.Z%MagicNum.ServerChunkSize, Pos.dimension);
+                if (inChunkPos.X is 0 or 31 || inChunkPos.Y is 0 or 31 || inChunkPos.Z is 0 or 31)
+                {
+                    (api.World as ServerMain)?.TriggerNeighbourBlocksUpdate(Pos);
+                    pnm.GetNetwork(NetworkID).MarkNetworkDirty(api.World); // no sure if it is the best to do here but resolve the issue with pipe not exporting when on multiple chunks
```

Comment:
MarkNetworkDirty should not be used unless you know it MUST be used to solve a problem, it's a VERY expensive call to make and should not be thrown around, especially if you are 'not sure if it the best to do here'.